src/common/actions/snap-builds.js

```diff
@@ -50,39 +63,52 @@ export function fetchSnap(repositoryUrl) {
           const snapLink = json.payload.message;
           dispatch({
             type: FETCH_SNAP_SUCCESS,
-            payload: snapLink
+            payload: {
+              id: fullName,
+              snapLink
+            }
           });
         })
-        .catch((error) => dispatch(fetchBuildsError(error)));
+        .catch((error) => dispatch(fetchBuildsError(fullName, error)));
     }
   };
 }
 
 // Fetch builds list for given snap
-export function fetchBuilds(snapLink) {
+export function fetchBuilds(repositoryUrl, snapLink) {
```

Comment:
We need both `repositoryUrl` and `snapLink` now to fetch builds, because we need to store them per-repo.

In theory `repositoryUrl` should be enough, as we should already have snapLink for it in the store, but I'm not sure if we want actions to read the store themselves.